packages/scripts/src/monitoring-dvm2.0/simulateVoting.ts

```diff
@@ -0,0 +1,663 @@
+    .transfer(foundationAddress, await votingToken.balanceOf(accountWithUma));
+  let foundationBalance = await votingToken.balanceOf(foundationAddress);
+
+  console.log(` 2. Foundation has ${formatEther(foundationBalance)} UMA, funding requester and voters...`);
+  // There will be 3 voters with initial balances set relative to GAT, and for two disputed price
+  // price requests 12 * finalFee amount will be needed (Optimistic Oracle bond defaults to finalFee).
+  if (
+    foundationBalance.lt(
+      gat
+        .mul(voter1RelativeGatFunding.add(voter2RelativeGatFunding.add(voter3RelativeGatFunding)))
+        .div(parseEther("1"))
+        .add(finalFee.mul(12))
+        .add(minimumStakedAmount)
+    )
+  )
+    throw new Error("Foundation balance too low for simulation!");
+
+  const [
+    requesterSigner,
+    voter1Signer,
+    voter2Signer,
+    voter3Signer,
+  ]: SignerWithAddress[] = await hre.ethers.getSigners();
+
+  let [requesterBalance, voter1Balance, voter2Balance, voter3Balance] = await Promise.all(
+    [requesterSigner, voter1Signer, voter2Signer, voter3Signer].map((signer) => {
+      return votingToken.balanceOf(signer.address);
+    })
+  );
+
+  // Transfering required balances. This assumes recipient accounts did not have more than target amounts before
+  // simulation.
+  await (
+    await votingToken
+      .connect(foundationSigner)
+      .transfer(requesterSigner.address, finalFee.mul(12).sub(requesterBalance))
+  ).wait();
+  await (
+    await votingToken
+      .connect(foundationSigner)
+      .transfer(voter1Signer.address, gat.mul(voter1RelativeGatFunding).div(parseEther("1").sub(voter1Balance)))
+  ).wait();
+  await (
+    await votingToken
+      .connect(foundationSigner)
+      .transfer(voter2Signer.address, gat.mul(voter2RelativeGatFunding).div(parseEther("1").sub(voter2Balance)))
+  ).wait();
+  await (
+    await votingToken
+      .connect(foundationSigner)
+      .transfer(voter3Signer.address, gat.mul(voter3RelativeGatFunding).div(parseEther("1").sub(voter3Balance)))
+  ).wait();
+
+  [requesterBalance, voter1Balance, voter2Balance, voter3Balance] = await Promise.all(
+    [requesterSigner, voter1Signer, voter2Signer, voter3Signer].map((signer) => {
+      return votingToken.balanceOf(signer.address);
+    })
+  );
+
+  console.log(` ✅ Requester now has ${formatEther(requesterBalance)} UMA.`);
+  console.log(` ✅ Voter 1 now has ${formatEther(voter1Balance)} UMA.`);
+  console.log(` ✅ Voter 2 now has ${formatEther(voter2Balance)} UMA.`);
+  console.log(` ✅ Voter 3 now has ${formatEther(voter3Balance)} UMA.`);
+
+  console.log(" 3. Staking minimum amount from foundation...");
+  // Get existing stakers that will be needed in the next step before foundation stakes.
+  const uniqueVoters = await getUniqueVoters(votingV2);
+
+  await votingToken.connect(foundationSigner).approve(votingV2.address, minimumStakedAmount);
+  await votingV2.connect(foundationSigner).stake(minimumStakedAmount);
```

Comment:
Commenting this line would break the tests with an overflow/underflow error right?
Maybe we could document a bit this behaviour here. Wdyt?